code/modules/atmospherics/machinery/components/binary_devices/pump.dm

```diff
@@ -180,3 +189,93 @@
+
+	///Pressure of the input port
+	var/datum/port/output/input_pressure
+	///Pressure of the output port
+	var/datum/port/output/output_pressure
+	///Temperature of the input port
+	var/datum/port/output/input_temperature
+	///Temperature of the output port
+	var/datum/port/output/output_temperature
+
+	///Whether the pump is currently active
+	var/datum/port/output/is_active
+	///Send a signal when the pump is turned on
+	var/datum/port/output/turned_on
+	///Send a signal when the pump is turned off
+	var/datum/port/output/turned_off
+
+	///The component parent object
+	var/obj/machinery/atmospherics/components/binary/pump/connected_pump
+
+/obj/item/circuit_component/atmos_pump/populate_ports()
+	pressure_value = add_input_port("New Pressure", PORT_TYPE_NUMBER, trigger = .proc/set_pump_pressure)
+	on = add_input_port("Turn On", PORT_TYPE_SIGNAL)
+	off = add_input_port("Turn Off", PORT_TYPE_SIGNAL)
+	request_data = add_input_port("Request Port Data", PORT_TYPE_SIGNAL)
+
+	input_pressure = add_output_port("Input Pressure", PORT_TYPE_NUMBER)
+	output_pressure = add_output_port("Output Pressure", PORT_TYPE_NUMBER)
+	input_temperature = add_output_port("Input Temperature", PORT_TYPE_NUMBER)
+	output_temperature = add_output_port("Output Temperature", PORT_TYPE_NUMBER)
+
+	is_active = add_output_port("Active", PORT_TYPE_NUMBER)
+	turned_on = add_output_port("Turned On", PORT_TYPE_SIGNAL)
+	turned_off = add_output_port("Turned Off", PORT_TYPE_SIGNAL)
+
+/obj/item/circuit_component/atmos_pump/register_usb_parent(atom/movable/shell)
+	. = ..()
+	if(istype(shell, /obj/machinery/atmospherics/components/binary/pump))
+		connected_pump = shell
+		RegisterSignal(connected_pump, COMSIG_PUMP_SET_ON, .proc/handle_pump_activation)
+
+/obj/item/circuit_component/atmos_pump/unregister_usb_parent(atom/movable/shell)
+	UnregisterSignal(connected_pump, COMSIG_PUMP_SET_ON)
+	connected_pump = null
+	return ..()
+
+/obj/item/circuit_component/atmos_pump/proc/handle_pump_activation(datum/source, active)
+	is_active.set_output(active)
+	if(active)
+		turned_on.set_output(COMPONENT_SIGNAL)
+	else
+		turned_off.set_output(COMPONENT_SIGNAL)
+
+/obj/item/circuit_component/atmos_pump/proc/set_pump_pressure()
+	CIRCUIT_TRIGGER
+	if(!connected_pump)
+		return
+	connected_pump.target_pressure = pressure_value.value
+
+/obj/item/circuit_component/atmos_pump/input_received(datum/port/input/port, list/return_values)
+
+	if(!connected_pump)
+		return
+
+	if(on == port)
+		connected_pump.set_on(TRUE)
+		return
+	if(off ==  port)
+		connected_pump.set_on(FALSE)
+		return
+	if(request_data == port)
+		var/datum/gas_mixture/air_input = connected_pump.airs[1]
+		var/datum/gas_mixture/air_output = connected_pump.airs[2]
+		input_pressure.set_output(air_input.return_pressure())
+		output_pressure.set_output(air_output.return_pressure())
+		input_temperature.set_output(air_input.return_temperature())
+		output_temperature.set_output(air_output.return_temperature())
```

Comment:
you can move these all to their own triggers as well. It's ideal if everything isn't in input_received

Reply:
aight!